pkg/clients/iam/iam_client.go

```diff
@@ -145,3 +152,88 @@ func (client *IAMClient) CreateBucketAccess(ctx context.Context, userName, bucke
 
 	return accessKeyOutput, nil
 }
+
+// RevokeBucketAccess is a helper that revokes bucket access by orchestrating individual steps to delete the user, inline policy, and access keys.
+func (client *IAMClient) RevokeBucketAccess(ctx context.Context, userName, bucketName string) error {
```

Comment:
So I assume there is a 1-1 mapping between users and buckets, in which case it looks correct to me.

Reply:
There is a 1 to many mapping for buckets and users.
1 bucket can have multiple IAM users accessing it.

But for each access indeed we have 1 IAM user